### PR TITLE
confirm-order endpoint

### PR DESCRIPTION
**Pull request by PDrago55**
Screen Shot 2020-04-11 at 12 24 57 PM

In order for the FE to have a successful POST, the following information is required for the body.. 
POST `/order`
```
{
  "order_summary": [
    {
      "item_id": 6546,
      "quantity": 1
    },
    {
      "item_id": 6890,
      "quantity": 16
    },
    {
      "item_id": 7037,
      "quantity": 2
    }
  ]
}
```
The ITEM_ID and QUANTITY need to be passed to the BE :)

### CODE REVIEW

**Review by dani-santos-code — CHANGES_REQUESTED** [review]
Hey! I was thinking...and in the end it's best to have validation in the backend for security purposes. Other than that, a hacker can attack us and order an amount that exceeds our stock items and break our service. :(

File: server/index.js [hunk 1]
```diff
@@ -93,6 +101,22 @@ express()
...
 
+  //Order-Form Validation
+
+  .post("/order", (req, res) => {
+    const { order_summary } = req.body;
+    const updateOrder = order_summary.map((item) => {
+      return productData
+        .filter((product) => product.id === item.item_id)
+        .map((orderItem) => {
+          orderItem.numInStock -= item.quantity;
```
Comment (dani-santos-code) [inline server/index.js]: So, here we would go back to the previous check we did...😬

```
if(orderItem.numInStock - item.id > 0) {
   orderItem.numInStock -= item.quantity;
  res.send({ status: "success" });
} else {
 res.send({ status: "error" });
}
```

**Review by JeanLoupBrault — APPROVED** [review]
confirm-order endpoint approved by Jean-Loup

File: server/data/order.json [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    {
+      "item_id": 6890,
+      "quantity": 16
+    },
+    {
+      "item_id": 7037,
+      "quantity": 2
+    }
+  ]
+}
```
Comment (dani-santos-code) [inline server/data/order.json:19]: We don't need to commit this. This was just for our testing purposes.

**Review by dani-santos-code — APPROVED** [review]

**Comment by PDrago55** [comment]
:)

**Pull request merged**